client/web/src/insights/pages/dashboards/dashboard-page/components/add-insight-modal/hooks/use-reachable-insights.ts

```diff
@@ -79,5 +86,29 @@ export function useReachableInsights(props: UseReachableInsightsProps): Reachabl
                         owner: subjectOwnerInfo,
                     }
                 })
+
+            const backendInsightSettings =
+                (settings?.[INSIGHTS_ALL_REPOS_SETTINGS_KEY] as Record<string, SearchBasedInsightSettings>) ?? {}
+
+            const backendBasedInsights = Object.keys(backendInsightSettings)
+                .filter(isInsightSettingKey)
+                .map(key => {
+                    const insightConfiguration = backendInsightSettings[key]
+
+                    const insight: Insight = {
+                        type: InsightType.Backend,
+                        id: key,
+                        visibility: ownerId,
+                        ...insightConfiguration,
+                    }
+
+                    return {
+                        ...insight,
+                        // Extend common insight object with owner info
+                        owner: subjectOwnerInfo,
+                    }
```

Comment:
```suggestion
                    const insight: Insight = {
                        type: InsightType.Backend,
                        id: key,
                        visibility: ownerId,
                        ...insightConfiguration,
                        // Extend common insight object with owner info
                        owner: subjectOwnerInfo,
                    }
                    return insight
```

(optional) Out of curiousity, is there any benefit to double spreading here?

Reply:
No, no special reasons I just missed that. Thanks, I will fix that and leave only one object definition here. 👍